jwst/extract_1d/extract.py

```diff
@@ -338,50 +338,45 @@ def get_extract_parameters(
                     # Note: extract_params['dispaxis'] is not assigned. This is done later, possibly slit by slit.
 
                     if meta.target.source_type == "EXTENDED":
-                        log.info("Target is extended, so the entire region will be extracted.")
                         shape = input_model.data.shape
-                        extract_params['src_coeff'] = None
-                        extract_params['bkg_coeff'] = None
-                        extract_params['bkg_fit'] = None
-                        extract_params['bkg_order'] = 0
-                        extract_params['subtract_background'] = False
-                        extract_params['use_source_posn'] = False
-                        extract_params['xstart'] = 0
-                        extract_params['xstop'] = shape[-1] - 1
-                        extract_params['ystart'] = 0
-                        extract_params['ystop'] = shape[-2] - 1
-                        extract_params['extract_width'] = None
-                        extract_params['independent_var'] = 'pixel'
-                        extract_params['position_correction'] = 0  # default value
+                        extract_params['xstart'] = aper.get('xstart', 0)
```

Comment:
What's the 2nd argument of the `aper.get()` function? The default value to use if the attribute doesn't exist?

Reply:
Yes, this will return 0 if `xstart` is not in the aper dict.